### PR TITLE
Switch doogie version

### PR DESCRIPTION
**Pull request by windwardrail**
This PR introduces internal changes to Howser resulting from updating the Doogie dependency to 1.0.0.

### CODE REVIEW

File: src/data.rs [hunk 1]
```diff
@@ -124,34 +123,27 @@ pub enum ElementType {
 
 impl ElementType {
     pub fn determine(node: &Node) -> HowserResult<Self> {
```
Comment (mullr) [inline src/data.rs]: Looks like this doesn't need to be a Result anymore, as it only ever returns Ok

File: src/document.rs [hunk 2]
```diff
@@ -9,117 +8,14 @@ use constants::{DITTO_TOKEN, MANDATORY_PROMPT, OPTIONAL_PROMPT, PROMPT_PATTERN};
 use data::ElementType;
 use data::{MatchType, NodeData};
 use doogie::constants::*;
-use doogie::Node;
+use doogie::{
+    Node,
+};
```
Comment (mullr) [inline src/document.rs]: ??

File: src/document.rs [hunk 3]
```diff
@@ -178,23 +61,16 @@ impl<'a> Document<'a> {
...
-            .ok_or(HowserError::CapabilityError)?;
-        let id = getter.get_id()?;
+        let id = node.get_id();
         let mut data_store = self.data.borrow_mut();
         let node_data = data_store.entry(id).or_insert(NodeData::new());
         Ok(node_data.match_type.clone())
     }
 
     /// Set the `MatchType` for a `Node`.
     fn set_match_type(&self, node: &Node, match_type: MatchType) -> Result<&Self, HowserError> {
```
Comment (mullr) [inline src/document.rs]: Result not needed

File: src/document.rs [hunk 4]
```diff
@@ -204,27 +80,21 @@ impl<'a> Document<'a> {
 
     /// Returns a boolean indicating if this `Node` is considered a wildcard.
     pub fn is_wildcard(&self, rx_node: &Node) -> HowserResult<bool> {
```
Comment (mullr) [inline src/document.rs]: result not needed

File: src/document.rs [hunk 5]
```diff
@@ -204,27 +80,21 @@ impl<'a> Document<'a> {
...
 
     /// Set the wildcard status for a `Node`.
-    fn set_is_wildcard(&self, node: &Node, state: bool) -> Result<&Self, HowserError> {
-        if let Some(ref getter) = node.capabilities.get {
-            let id = getter.get_id()?;
-            let mut data = self.data.borrow_mut();
-            let node_data = data.entry(id).or_insert(NodeData::new());
-            node_data.is_wildcard = state;
-        }
+    fn set_is_wildcard(&self, node: &Node, state: bool) -> HowserResult<()> {
```
Comment (mullr) [inline src/document.rs]: ResultNotNeeded

File: src/document.rs [hunk 6]
```diff
@@ -693,87 +416,42 @@ fn get_code_block_annotation(node: &Node) -> HowserResult<MatchType> {
...
+    if let Some(text_node) = heading_node.first_child()? {
+        if let Node::Text(ref text) = text_node {
+            if text_node.next_sibling()?.is_none() {
+                let (match_type, content) = extract_match_type(&text.get_content()?)?;
                 if match_type != MatchType::None && content.is_empty() {
-                    Ok(match_type)
+                    return Ok(match_type);
                 } else {
-                    Ok(MatchType::None)
+                    return Ok(MatchType::None);
```
Comment (mullr) [inline src/document.rs]: Why did you choose to add the returns here? I was about to comment about how they're not needed, but I see you added them in this change.

File: src/document.rs [hunk 7]
```diff
@@ -693,87 +416,42 @@ fn get_code_block_annotation(node: &Node) -> HowserResult<MatchType> {
...
-            let getter = text.capabilities
-                .get
-                .as_ref()
-                .ok_or(HowserError::CapabilityError)?;
-            if getter.get_type()? == NodeType::CMarkNodeText {
-                let (match_type, content) = extract_match_type(&getter.get_content()?)?;
+fn get_heading_annotation(heading_node: &Node) -> HowserResult<MatchType> {
+    if let Some(text_node) = heading_node.first_child()? {
+        if let Node::Text(ref text) = text_node {
+            if text_node.next_sibling()?.is_none() {
```
Comment (mullr) [inline src/document.rs]: This nesting is getting kind of deep; I wonder if you can flatten it by inverting the conditions and returning early? Might not work with the `if let` idiom.

File: src/document.rs [hunk 8]
```diff
@@ -693,87 +416,42 @@ fn get_code_block_annotation(node: &Node) -> HowserResult<MatchType> {
...
-            let getter = text.capabilities
-                .get
-                .as_ref()
-                .ok_or(HowserError::CapabilityError)?;
-            if getter.get_type()? == NodeType::CMarkNodeText {
-                let (match_type, content) = extract_match_type(&getter.get_content()?)?;
+fn remove_heading_annotation(heading_node: &Node) -> HowserResult<()> {
+    if let Some(text_node) = heading_node.first_child()? {
+        if let Node::Text(ref text) = text_node {
+            if text_node.next_sibling()?.is_none() {
```
Comment (mullr) [inline src/document.rs]: some comment w.r.t. depth

File: src/validator.rs [hunk 9]
```diff
@@ -9,8 +9,9 @@ use constants::{CONTENT_PROMPT_PATTERN, MANDATORY_PROMPT, OPTIONAL_PROMPT};
 use data::ElementType;
 use data::{ContentMatchPair, MatchType, PromptToken};
 use document::{Document, Prescription};
-use doogie::constants::NodeType;
-use doogie::Node;
+use doogie::{
+    Node,
+};
```
Comment (mullr) [inline src/validator.rs]: Again, ??

File: src/document.rs (hunk 6 above)
Comment (windwardrail) [reply src/document.rs]: You are right. The second early return was not needed

File: src/document.rs [hunk 10]
```diff
@@ -922,12 +546,97 @@ fn strip_comments(root: &Node) -> HowserResult<()> {
     Ok(())
 }
 
+/// Abstraction of the different types of block level Rx Elements and their match context.
```
[the remark this answers is not in the document]
Comment (windwardrail) [reply src/document.rs]: I believe so, yes

File: src/document.rs [hunk 11]
```diff
@@ -844,22 +489,13 @@ fn remove_item_annotation(node: &Node) -> HowserResult<()> {
...
 fn annotate_circumstantial_node(target_node: &Node, document: &Document) -> HowserResult<()> {
     document.set_match_type(target_node, MatchType::Optional)?;
-    let traverser = target_node
-        .capabilities
-        .traverse
-        .as_ref()
-        .ok_or(HowserError::CapabilityError)?;
-    let mut child = traverser.first_child()?;
+    let mut child = target_node.first_child()?;
     while let Some(node) = child {
```
[the remark this answers is not in the document]
Comment (windwardrail) [reply src/document.rs]: This is a great idea, but I'm going to defer it to a different PR since it involves adding functionality to doogie first.

**Comment by windwardrail** [comment]
Comments addressed @mullr , Thanks!

**Pull request merged**